spock-specs/src/test/groovy/org/spockframework/docs/interaction/StaticMocksDocSpec.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.docs.interaction
+
+
+import spock.lang.Specification
+import spock.mock.IStaticMock
+
+class StaticMocksDocSpec extends Specification {
+
+  def "StaticMock"() {
+    // tag::mock-static1[]
+    given:
+    MockStatic(StaticClass)
+    expect: "By default MockStatic() will return null"
+    StaticClass.staticMethod() == null
+
+    when:
+    def result = StaticClass.staticMethod()
+    then:
+    1 * StaticClass.staticMethod() >> "MockValue"
+    result == "MockValue"
+    // end::mock-static1[]
+  }
+
+  def "StaticMock with answers"() {
+    // tag::mock-static-answers[]
+    given:
+    MockStatic(StaticClass) {
+      staticMethod() >> "MockValue"
+    }
+    expect:
+    StaticClass.staticMethod() == "MockValue"
+    // end::mock-static-answers[]
+  }
+
+  def "StaticMock with interactions"() {
+    // tag::mock-static-interactions[]
+    given:
+    MockStatic(StaticClass)
+    StaticClass.staticMethod() >> "MockValue"
+    expect:
+    StaticClass.staticMethod() == "MockValue"
+    // end::mock-static-interactions[]
+  }
+
+  def "SpyStatic"() {
+    // tag::spy-static[]
+    given:
+    SpyStatic(StaticClass)
+    expect:
```

Comment:
❌ Add a newline before the block labels when starting a new block, to increase readability.